nautobot/dcim/tests/test_views.py

```diff
@@ -2223,6 +2223,19 @@ def setUpTestData(cls):
             "controller_managed_device_group": ControllerManagedDeviceGroup.objects.first().pk,
         }
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_vdc_panel_includes_add_vdc_btn(self):
+        """Assert Add Virtual device Contexts button is in Device detail view: Issue from #6348"""
+        device = Device.objects.first()
+        url = reverse("dcim:device", kwargs={"pk": device.pk})
+        response = self.client.get(url)
+        response_context = response.content.decode(response.charset)
+        vdc_url = reverse("dcim:virtualdevicecontext_add")
+        return_url = device.get_absolute_url()
+        vdcs_table_add_url = f"{vdc_url}?device={device.id}&amp;return_url={return_url}"
```

Comment:
I would move this section above the actual call to the view and rename to `expected_vdcs_table_add_url` to increase the readability. Also adding a new line separator between setup data, actual test and asserting output might be a good idea :)